src/components/FallbackUIButton/index.tsx

```diff
@@ -1,31 +1,19 @@
-import React, { Component } from 'react';
-
-import { FallbackUIButtonState } from './types/types';
+import React, { useState } from 'react';
 import { Button } from '../../styles';
 
-export default class FallbackUIButton extends Component<
-  NonNullable<unknown>,
-  FallbackUIButtonState
-> {
-  constructor(props: NonNullable<unknown>) {
-    super(props);
-    this.state = { isError: false };
-  }
-
-  render() {
-    const { isError } = this.state;
+export default function FallbackUIButton() {
+  const [isError, setIsError] = useState<boolean>(false);
 
-    let content: JSX.Element;
+  let content: JSX.Element;
 
-    switch (true) {
-      case isError:
-        throw new Error('Test error throwing');
+  switch (true) {
```

Comment:
It looks strange, please look for another way to draw different states of the component.